packages/openapi2-parser/lib/parser.js

```diff
@@ -1611,7 +1611,11 @@ class Parser {
     const referencedPathValue = this.referencedPathValue();
     let cacheKey;
     if (referencedPathValue && referencedPathValue.$ref) {
+      // schema object with $ref
       cacheKey = `${referencedPathValue.$ref};${contentType}`;
+    } else if (referencedPathValue && referencedPathValue.allOf && Object.keys(referencedPathValue).length === 1 && referencedPathValue.allOf.length === 1 && referencedPathValue.allOf[0].$ref) {
```

Comment:
I'd wrap the code somehow so that reviewing is easier (ideally no horizontal scrolling).